### PR TITLE
[ Refactor ] Layer Method for weight initialization

### PR DESCRIPTION
**Pull request by jijoongmoon**
Cannot use weight initiatlization method for other layer.
In this PR, Move it into Layer class to use for other layer.

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: nntrainer/include/layer.h [hunk 1]
```diff
@@ -194,18 +194,54 @@ class Layer {
...
+  /**
+   * @brief  set Weight Initialization Type
+   * @param[in] wini WeightIniType
+   */
   void setWeightInit(WeightIniType wini) { weight_ini_type = wini; }
 
+  /**
+   * @brief  initialize Weight
+   * @param[in] width width of Tensor
+   * @param[in] width height of Tensor
```
Comment (dongju-chae) [inline nntrainer/include/layer.h:237]: Typo?

File: nntrainer/include/layer.h [hunk 2]
```diff
@@ -194,18 +194,54 @@ class Layer {
    */
   virtual void copy(std::shared_ptr<Layer> l) = 0;
 
+  /**
+   * @brief     set Batch Normalization Layer followed
+   * @param[in] ok true/false
+   */
   void setBNfallow(bool ok) { this->bn_fallow = ok; }
```
Comment (zhoonit) [inline nntrainer/include/layer.h:201]:
```suggestion
  void setBNfollow(bool ok) { this->bn_follow = ok; }
```
Is this typo? If it is, I might work on it.

File: nntrainer/include/layer.h (hunk 1 above)
Comment (jijoongmoon) [reply nntrainer/include/layer.h]: fixed.

File: nntrainer/include/layer.h (hunk 2 above)
Comment (jijoongmoon) [reply nntrainer/include/layer.h]: fixed. Thanks.

**Pull request merged**